shared/services/config/smartnode-config.go

```diff
@@ -804,113 +530,105 @@ func (cfg *SmartnodeConfig) GetFeeRecipientFilePath() string {
 }
 
 func (cfg *SmartnodeConfig) GetV100RewardsPoolAddress() common.Address {
-	return common.HexToAddress(cfg.v1_0_0_RewardsPoolAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_0_0_RewardsPool)
 }
 
 func (cfg *SmartnodeConfig) GetV100ClaimNodeAddress() common.Address {
-	return common.HexToAddress(cfg.v1_0_0_ClaimNodeAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_0_0_ClaimNode)
 }
 
 func (cfg *SmartnodeConfig) GetV100ClaimTrustedNodeAddress() common.Address {
-	return common.HexToAddress(cfg.v1_0_0_ClaimTrustedNodeAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_0_0_ClaimTrustedNode)
 }
 
 func (cfg *SmartnodeConfig) GetV100MinipoolManagerAddress() common.Address {
-	return common.HexToAddress(cfg.v1_0_0_MinipoolManagerAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_0_0_MinipoolManager)
 }
 
 func (cfg *SmartnodeConfig) GetV110NetworkPricesAddress() common.Address {
-	return common.HexToAddress(cfg.v1_1_0_NetworkPricesAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_1_0_NetworkPrices)
 }
 
 func (cfg *SmartnodeConfig) GetV110NodeStakingAddress() common.Address {
-	return common.HexToAddress(cfg.v1_1_0_NodeStakingAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_1_0_NodeStaking)
 }
 
 func (cfg *SmartnodeConfig) GetV110NodeDepositAddress() common.Address {
-	return common.HexToAddress(cfg.v1_1_0_NodeDepositAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_1_0_NodeDeposit)
 }
 
 func (cfg *SmartnodeConfig) GetV110MinipoolQueueAddress() common.Address {
-	return common.HexToAddress(cfg.v1_1_0_MinipoolQueueAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_1_0_MinipoolQueue)
 }
 
 func (cfg *SmartnodeConfig) GetV110MinipoolFactoryAddress() common.Address {
-	return common.HexToAddress(cfg.v1_1_0_MinipoolFactoryAddress[cfg.Network.Value.(config.Network)])
+	return common.HexToAddress(cfg.GetNetworkInfo().Addresses.V1_1_0_MinipoolFactory)
 }
 
 func (cfg *SmartnodeConfig) GetPreviousRewardsPoolAddresses() []common.Address {
-	return cfg.previousRewardsPoolAddresses[cfg.Network.Value.(config.Network)]
+	addresses := []common.Address{}
```

Comment:
incredibly premature optimization from me:
```suggestion
	addresses := make([]common.Address, 0, len(cfg.GetNetworkInfo().Addresses.PreviousRewardsPools))
```